consensus/consensus_v2.go

```diff
@@ -606,12 +606,23 @@ func (consensus *Consensus) preCommitAndPropose(blk *types.Block) error {
 		// Send signal to Node to propose the new block for consensus
 		consensus.getLogger().Info().Msg("[preCommitAndPropose] sending block proposal signal")
 		consensus.mutex.Unlock()
-		consensus.ReadySignal(NewProposal(AsyncProposal), "preCommitAndPropose", "proposing new block which will wait on the full commit signatures to finish")
+		if !consensus.isRotation(blk.Epoch()) {
+			consensus.ReadySignal(NewProposal(AsyncProposal, blk.NumberU64()+1), "preCommitAndPropose, before rotation", "proposing new block which will wait on the full commit signatures to finish")
+		} else {
+			next := consensus.rotateLeader(blk.Epoch(), consensus.getLeaderPubKey())
+			if consensus.isMyKey(next) {
+				consensus.ReadySignal(NewProposal(AsyncProposal, blk.NumberU64()+1), "preCommitAndPropose rotation", "proposing new block which will wait on the full commit signatures to finish")
+			}
+		}
 	}()
 
 	return nil
 }
 
+func (consensus *Consensus) isRotation(epoch *big.Int) bool {
```

Comment:
why re-creating a new function and not use directly IsLeaderRotationInternalValidators ?

Reply:
i don't need to know which kind of rotation, i need the fact.

Reply:
ok and why recreating the function ?

Reply:
because leader rotation does not affect blockchain anyhow. It means, that we can safely remove internal and external epochs, and left only v2. Exactly in this part of code, i don't need to know which kind of rotation we use, i need to know that we have it active.